prov/shm/src/smr_av.c

```diff
@@ -112,6 +110,9 @@ static int smr_av_insert(struct fid_av *av_fid, const void *addr, size_t count,
 			smr_av->used++;
 		}
 
+		if (fi_addr)
+			fi_addr[i] = util_addr;
```

Comment:
There's a continue case above that skips this assignment.  nit: There's an else statement that follows the continue which can be removed.

Reply:
> There's a continue case above that skips this assignment.

Yes, I observed that too. That means we were assigning this addr always, no matter the assertion succeed or not. Was that legitimate?

Reply:
Yes, the util insert call will return FI_ADDR_NOTAVAIL if it was unassigned which is the behavior we want I think

Reply:
I see, I will get that assignment back in this case. Thanks!